src/components/main-view/main-view.jsx

```diff
@@ -1,77 +1,119 @@
-          description: movie.Description, // Description field from the DB
+          id: movie._id,
+          title: movie.Title,
+          image: movie.ImageUrl,
+          director: movie.Director,
+          genre: movie.Genre,
+          description: movie.Description,
         }));
-        setMovies(moviesFromApi); // Set the movies state
+
+        setMovies(moviesFromApi);
       });
-  }, [token]); // Depend on token to fetch movies when it changes
+  }, [token]);
+
+  return (
+    <Row className="justify-content-md-center">
+      {!user ? (
+        
+        <Col md={5}>
+          <LoginView onLoggedIn={(user, token) => {
+            setUser(user);
+            setToken(token);
+          }} />
+          <SignupView />
+          </Col>
+        
+      ) : selectedMovie ? (
+        <Col md={8} style={{ border: "1px solid black" }}>
+        <MovieView movie={selectedMovie} onBackClick={() => setSelectedMovie(null)} />
+        </Col>
+      ) : movies.length === 0 ? (
+        <div>The list is empty!</div>
+      ) : (
+        <>
+          {movies.map((movie) => (
+            <Col className="mb-5" key={movie.id} md={3}>
+            <MovieCard key={movie.id} movie={movie} onMovieClick={() => setSelectedMovie(movie)} /> </Col>
+          ))}
+        </>
+      )}
+    </Row>
+  );
+};
 
-  // Function to handle logout
-  const handleLogout = () => {
-    setUser(null);
-    setToken(null);
-    localStorage.removeItem("user");
-    localStorage.removeItem("token");
+/*
+  // Function to handle the back click
+  const handleBackClick = () => {
+    setSelectedMovie(null); // Reset selected movie to null to go back to the main view
   };
 
-  // If no user is logged in, show the login view
   if (!user) {
     return (
-      <LoginView
-        onLoggedIn={(user, token) => {
-          setUser(user); // Set the logged-in user
-          setToken(token); // Set the token
-          localStorage.setItem("user", JSON.stringify(user)); // Save user to localStorage
-          localStorage.setItem("token", token); // Save token to localStorage
-        }}
-      />
+      <>
+        <LoginView onLoggedIn={(user, token) => {
+          setUser(user);
+          setToken(token);
+        }} />
+        or
+        <SignupView />
+      </>
     );
   }
 
   // If a movie is selected, show the MovieView
   if (selectedMovie) {
-    return <MovieView movie={selectedMovie} onBackClick={() => setSelectedMovie(null)} />;
+    return <MovieView movie={selectedMovie} onBackClick={handleBackClick} />;
   }
 
   // If no movies are fetched, show a message
   if (movies.length === 0) {
     return <div>The list is empty!</div>;
   }
 
+  useEffect(() => {
+    if (!token) {
+      return;
+    }
+
+    fetch("https://movie-api-lvgy.onrender.com/movies", {
+      headers: { Authorization: `Bearer ${token}` }
+    })
+      .then((response) => response.json())
+      .then((data) => {
+        console.log(data);
+      });
+  }, [token]);
+
```

Comment:
Hi @allmyprogramming ,
It's best practice to avoid including commented-out lines of code, as they can make your source code less tidy.